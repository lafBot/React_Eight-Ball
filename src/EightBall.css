#EightBall {
    border-radius: 50%;
    height: 500px;
    width: 500px;
    position: relative;
    left: auto;
    right: auto;
    box-shadow: 100px 20px 40px;
    display: flex;
    justify-content: center;
    align-items: center;
    text-align: center;
}

#EightBall-text-container {
    background-color: rgba(255,255,255,0.2);
    padding: 10px;
    border-radius: 15px;
    min-width: 300px;
}

button {
    border-radius: 15px;
    padding: 10px;
    background-color: rgba(255,255,255,0.5);
}