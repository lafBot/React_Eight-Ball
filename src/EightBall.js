// Magic eight ball components
import React, { useState } from 'react';
import setAnswer from './setAnswer';
import './EightBall.css';

const EightBall = () => {
    /* Handles setting of color and msg state */
    const [msg, setMsg] = useState('Think of a Question')
    const [color, setColor] = useState('black');

    /* Calls setAnswer() to retrieve random msg and color combination
            then calls state functions to set color and msg*/
    function handleClick() {
        const { msg, color } = setAnswer();
        setColor(color);
        setMsg(msg);
    }
    // Returns div containing EightBall components
    return (
        <div id='EightBall' style={{backgroundColor: color}}>
            <div id='EightBall-text'>
                <h1>{msg}</h1>
                <button onClick={handleClick}>Shake</button>
            </div>
        </div>
    )
}

export default EightBall;