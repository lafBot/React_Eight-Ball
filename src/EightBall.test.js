import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'
import EightBall from './EightBall';

it('should render without crashing', () => {
    render(<EightBall />);
})

it('should match snapshot', () => {
    const {asFragment} = render(<EightBall />);
    expect(asFragment()).toMatchSnapshot();
})

it('should render new EightBall response', () => {
    const { getByText } = render(<EightBall />);
    let advice = getByText('Think of a Question');
    expect(advice).toBeInTheDocument();
    expect(advice).toHaveClass('EightBall-text');
    fireEvent.click(getByText('Shake'));
    expect(advice).not.toHaveTextContent('Think of a Question')
})