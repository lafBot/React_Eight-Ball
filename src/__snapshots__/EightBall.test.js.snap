// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    id="EightBall"
    style="background-color: black;"
  >
    <div
      id="EightBall-text-container"
    >
      <h1
        class="EightBall-text"
      >
        Think of a Question
      </h1>
      <button>
        Shake
      </button>
    </div>
  </div>
</DocumentFragment>
`;
